### DISCUSSION TITLE
NestJs | Vercel | CORS No 'Access-Control-Allow-Origin' header is present on the requested resource

### DISCUSSION THREAD

**Discussion by killswitchh** [vercel/community#383] [Help] [answered]
I have to deploy a nestJs backend to vercel.
Here is my vercel.json file
```
{
    "version": 2,
    "env": {
        "DATABASE_URL": "mongodb+srv://<mongourl>"
    },
    "builds": [
        {
            "src": "src/main.ts",
            "use": "@vercel/node"
        }
    ],
    "routes": [
        {
            "src": "/(.*)",
            "dest": "src/main.ts",
            "methods": [
                "GET",
                "POST",
                "PUT",
                "DELETE"
            ],
            "headers": {
                "Access-Control-Allow-Credentials": "true",
                "Access-Control-Allow-Origin": "*",
                "Access-Control-Allow-Headers": "X-CSRF-Token, X-Requested-With, Accept, Accept-Version, Content-Length, Content-MD5, Content-Type, Date, X-Api-Version"
            }
        }
    ]
}
```
I'm able to query the endpoints and get the response from Postman.
However when I'm trying to connect my Angular app to this end point its throwing me this error
Access to XMLHttpRequest at 'https://backend-url/users/email/a' from origin 'http://localhost:4200' has been blocked by CORS policy: No 'Access-Control-Allow-Origin' header is present on the requested resource.  
However, I have this code in my backend that should allow all origins
main.ts
```
import { NestFactory } from '@nestjs/core';
import { AppModule } from './app.module';

async function bootstrap() {
  const app = await NestFactory.create(AppModule);
  app.enableCors({
    origin: true,
    methods: 'GET,HEAD,PUT,PATCH,POST,DELETE,OPTIONS',
    credentials: true,
  });
  await app.listen(3000);
}

bootstrap();
```
Any help would be appreciated thanks! I tried asking the same in StackOverflow and got downvoted with no responses

**Comment by swarnava** [comment, marked as answer]
Hey,

Have you tried this? https://docs.nestjs.com/security/cors. You probably can get more help from the nest.js community: https://stackoverflow.com/questions/tagged/nestjs

**Reply by killswitchh** [reply]
Hi, 

I did try that, the implementation I have uses ```app.enableCors()``` . Thanks for pinging the nestJs community, I'll look for answers from them.

**Reply by siamahnaf** [reply]
hello, are you get the solution, I am getting the same error?
